clientv1_sessions.go

```diff
@@ -35,3 +37,19 @@ func (s *SessionsServiceV1) GetSessionByID(ctx context.Context, id string) (*cli
 	}
 	return resp.Msg.Session, nil
 }
+
+// SignOutSession revokes the authenticated state of the session with the given
+// ID for the given user. It does not return error if the session does not exist
+// or is not authenticated. It returns ErrRecordMismatch if the session is not
+// authenticated by the given user.
```

Comment:
@evict - Would it be too much to require the session is authenticated by the same SAMS user (in terms of security)? Assuming clients that are granted scope `sams::session::write` all already trusted (i.e. Sourcegraph-operated services).

Reply:
@unknwon yes, we must make sure we can't sign-out users unless the scope would allow them to. Forced sign-outs can help with phishing attacks for instance (they'd still have to guess the ID of a user, which is not trivial though).

Reply:
>There might be cases where we want to force sign-out users?

That sounds like an administrative operation? i.e. we can always provide that ability for admin/management APIs.

In the context of this session API, it seems worth starting with checking both the scope and the same user ID.